opm/input/eclipse/Schedule/MSW/WellSegments.cpp

```diff
@@ -616,6 +618,22 @@ namespace Opm {
         return segment.depth() - outlet_segment.depth();
     }
 
+    void WellSegments::checkSegmentDepthConsistency(const std::string& well_name) const {
+        for (const auto& segment : this->m_segments) {
+            const int segment_number = segment.segmentNumber();
+            if (segment_number == 1) {
+                continue; // not check the top segment for now
+            }
+            const double segment_length = this->segmentLength(segment_number);
+            const double segment_depth_change = this->segmentDepthChange(segment_number);
+            if (std::abs(segment_depth_change) > 1.001 * segment_length) { // 0.1% tolerance for comparison
+                const std::string msg = fmt::format(" Segment {} of well {} has a depth change of {} meters,"
+                                                    " while it has a length of {} meters, which is unphysical.",
```

Comment:
Talking about "meters" is going to look somewhat out of place if the run uses FIELD units (lengths in feet).  If there's no (easy) way to get a `UnitSystem` object here, then I suggest we rephrase it to use relative measures instead, e.g.,

> Depth change in segment {} of well {} exceeds total segment length by a factor of {}.

and then pass `abs(segment_depth_change) / segment_length` as the "factor".

Reply:
Let me investigate a little bit how to get the Unit System here, otherwise, I will go with your suggestion regarding using a factor instead.